[applicationForm] 동아리 가입 신청 답변 API 리팩토링

rugaheedori opened the pull request:
## 작업내용
1. 동아리 가입 신청 답변 작성 시 사용되는 로직 함수 추출
    1 )  중복 가입 신청 방지 => checkApplicantRecord / 동아리 회장 판별 추가
    2 )  전화 번호  형식 및 중복 체크 => phoneNumCheck
    3 )  필수 질문 답변 작성  => createBasicAnswer
    4 )  추가  질문 답변 작성 => createExtraAnswer
2. adminOption과 관련된 코드 삭제
3. DB query 컨벤션 수정
4. ctrl utils getApiInfo 및 proccessCtrl 사용하여 return 과 log 관리
5. Error.ctrl msg 삭제

## 주의사항
주의사항 내용

rugaheedori committed:
Refactor(gahee/application-form) : 동아리 가입 중복 신청 방지 로직 함수 추출 => isApplicant() 및 전화 번호 형식, 사용 중 확인 로직 함수 추출 => phoneNumCheck() #463

rugaheedori committed:
Refactor(gahee/application-form) : ApplicationStoreage query문 컨벤션 수정 #463

rugaheedori committed:
Delete(gahee/application-form) : adminOption과 관련된 코드 삭제 #463

rugaheedori committed:
Refacotor(gahee/application-form) : #463
1. 이전 신청 기록 조회 로직 함수 추출 => isApplicant
2. 전화 번호 형식 및 중복 사용 확인 로직 함수 추출 => phoneNumCheck
3. 필수 답변 작성 함수 추출 => createBasicAnswer
4. 추가 답변 작성 함수 추출 시 기존 답변 이력 삭제 로직 함수 추출 => deleteExtraAnswer
5. 추가 답변 작성 함수 추출 => createExtraAnswer
6. 답변 작성 성공 후 신청기록 추가 함수 추출 => createApplicant

rugaheedori committed:
Refactor(gahee/application-form) : 이전 신청 기록 조회 함수명 변경 isApplicant => checkApplicantRecord 및 변수명 변경 isApplicant => applicant #463

rugaheedori committed:
Refactor(gahee/application-form) : #463
1. createBasicAnswer 내 코드 순서 변경
2. phoneNumCheck 매개변수 값 추가
3. basicAnswerInfo에 불필요한 값 name 삭제
4. 답변 생성 확인 변수명 변경 isBasic => createBasicAnswer
5. 핸드폰 번호 중복 확인 변수명 변경 isPhoneNum => existPhoneNum

rugaheedori committed:
Modify(gahee/application-form) : phoneNumCheck 변수명 오타 수정 및 Storage query문 변수명 변경 #463

rugaheedori committed:
Refactor(gahee/application-form) : #463
1. 동아리 회장이 가입 신청한 경우 조건문 추가
2. 전화번호 중복 체크함수 ApplicationStorage의 fineDuplicatePhoneNum추가 및 변수명 existPhoneNum => duplicatePhoneNum으로 변경
3. ctrl return 값 및 log 메세지 utils의 getApiInfo, proceessCtrl 사용으로 변경
4. Error 메세지 빈값으로 변경
5. application.js return값 makeMsg함수 사용으로 변경

rugaheedori committed:
Refactor(gahee/application-form) : DB 쿼리 컨벤션 수정 #463

rugaheedori committed:
Refactor(gahee/application-form) : 직관적인 변수명으로 변경 #463

rrgks6221 reviewed (commented):
.js 8번 라인 this.req 지워주세요.
index.js 주석 제거해주세요.

rugaheedori committed:
Refactor(gahee/application-form) : ctrl 순서 index와 맞춰주기 #463

rugaheedori committed:
Refactor(gahee/application-form) : PHONE_NUMBER_REGEXP 값 변경, 전화번호 형식 확인 match => test로 변경 #463

rugaheedori committed:
Refactor(gahee/application-form) : 기존 답변 삭제 시 forEach()의 매개변수명 변경 x => extraAnswer #463

rugaheedori committed:
Refactor(gahee/applicariton-form) : createBasicAnswer내의 필수답변 빈칸 확인 부분 함수로 추출 => nullCheckBasicAnswer #463

rugaheedori committed:
Refactor(gahee/application-form) : 변수명 오타 수정 #463

rugaheedori committed:
Refactor(gahee/application-form) : #463
1. nullCheckBasicAnswer 함수 return값 Boolean 형태로 바꾸기 및 createAnswer 함수에서 makeMsg 사용
2. checkPhoneNum함수 checkPhoneNumFormat 함수와 checkDuplicastePhoneNum 함수로 분리
3. checkPhoneNumFormat 함수, checkDuplicatePhoneNum 함수 return 값 Boolean 형태로 바꾸기 및 createAnswer 함수에서 makeMsg 사용
4. nullCheckBasicAnswer 함수 this값이 아닌 basicAnswer 매개변수 사용
5. checkPhoneNumFormat 함수 this값이 아닌 basicAnswer.phoneNum 매개변수 사용

rugaheedori committed:
Refactor(gahee/application-form) : 서브 함수용 파일 ApplicationUtils.js 생성 및 makeMsg ApplicationUtils로 이동 및 적용 #453 #463

rugaheedori committed:
Refactor(gahee/application-form) : findAllByClubNum 서브 함수 findOneLeader/findOneClient/findAllQuestions ApplicationUtils로 이동 및 this 값 => 매개변수값 이용으로 변경 #453

rugaheedori committed:
Refactor(gahee/application-form) : updateQeustion, deleteQuestion 서브 함수 findOneWaitingApplicant 함수 ApplicationUtils로 이동 및 매개변수 적용 #453

rugaheedori committed:
Refactor(gahee/application-form) : 중복 신청 함수 checkApplicantRecord ApplicationUtils로 이동 및 this값 => 매개변수 변경 및 id => clientId 변수명 변경 #463

rugaheedori committed:
Refactor(gahee/application-form) : createAnswer 서브 함수들 ApplicationUtils 이동 및 매개변수 설정 및 변수명 변경 #463

rugaheedori committed:
Refactor(gahee/appliction-form) : fineOneClient ApplicantUtils에서 삭제 및 Storage로 연결 #453

rugaheedori committed:
Refactor(gahee/appliction-form) : findAllQuestions ApplicantUtils에서 삭제 및 Storage로 연결 #453

rugaheedori committed:
Refactor(gahee/appliction-form) : findOneWaitingApplicant  ApplicantUtils에서 삭제 및 Storage로 연결 #453

rugaheedori committed:
Refactor(gahee/application-form) : nullCheckBasicAnswer 함수 삭제 => 조건문에 return문 바로 사용 #463

rugaheedori committed:
Refactor(gahee/appliction-form) : checkApplicantRecord  ApplicantUtils에서 삭제 및 Storage로 연결 #463

rugaheedori committed:
Refactor(gahee/appliction-form) : checkDuplicatePhoneNum  ApplicantUtils에서 삭제 및 Storage findDuplicatePhoneNum로 연결 #463

rugaheedori committed:
Refactor(gahee/appliction-form) : deleteExtraAnswer => 질문 번호 추출부분 extractQuestionNums 함수로 추출 및 deleteExtraAnswer Storage로 연결 #463

rugaheedori committed:
Refactor(gahee/appliction-form) : createExtraAnswer, createApplicant ApplicantUtils에서 삭제 및 Storage로 연결 #463

rugaheedori committed:
Refactor(gahee/application-form) : extraQuestionNums 불필요한 변수 선언 삭제 #463

rugaheedori committed:
Refactor(gahee/application-form) : 많이 사용되는 ApplicationUitl.makeMsg 함수 makeMsg로 단순하게 선언 후 사용 #463
